schema.yaml

```diff
@@ -1,66 +1,110 @@
 ## Copyright © 2021, Oracle and/or its affiliates. 
 ## All rights reserved. The Universal Permissive License (UPL), Version 1.0 as shown at http://oss.oracle.com/licenses/upl
 
-title: Jboss/Wildfly Terraform Input Variables
-description: Jboss/Wildfly Terraform Input Variables
+title: Deploy WildFly connected to an autonomous database
+description: Deploy WildFly connected to an autonomous database
 schemaVersion: 1.1.0
 version: "20200821"
 variableGroups:
-  - title: "Network"
+  - title: General Configuration
+    visible: false
     variables:
-      - ${compartment_id}
-      - ${vcn_cidr_block}
-      - prefix
+    - tenancy_ocid
+    - region
+    - release
+    - public_subnet
 
-  - title: "Instances"
+  - title: Required Configuration
+    visible: true  
     variables:
-      - jboss_node_count
-      - jboss_vm_shape
-      - ssh_authorized_keys
-      - jboss_admin_username
-      - jboss_admin_password
-
-  - title: "Deployment Mode"
+    - compartment_id
+    - jboss_node_count
+    - jboss_admin_password
+    - atp_admin_password
```

Comment:
`atp_admin password` is not required unless the atp DB is provisioned, which is optional... I think this fits better with the ATP DB variable definitions as it was originally.

Reply:
Associated [Reference Architecture](https://docs.oracle.com/en/solutions/wildfly-oci/index.html) defines ATP as a part of the topology. That is why, just to be in sync with RA, I have added ATP_admin_password as required.

Reply:
I just tried your branch with oRM and I still find that the schema is not ideal: 
If ATP is provisioned, then the password is required. Right now, ATP is provisioned by default, but password is hidden and not required. It is burried under the advanced options. If ATP is provisioned then maybe advanced options need to show, or ATP password need to show on top. Either way it should be required or the stack will fail with an empty password.

The SSH key is also optional and in advanced options: we should encourage the standard practice of providing a SSH key, or the user will never be able to gain full access to the VM (unless they go through the complicated steps of resetting it via the boot manager).

It feels like we're trying to make it too simple, and in effect it gets more complicated to understand what went wrong if the stack fails.

Reply:
I guess it should be ok now. With the latest commit added to this PR chain I have introduced some changes. The checkbox for `provision_atp` makes `atp_admin_password` visible in basic options when flag is set to true. The `atp_admin_password` is only required when `provision_atp` is set to true. Also I have moved `ssh_authorized_keys` to basic options (not in advanced options anymore)